model/accessFile.js

```diff
@@ -0,0 +1,38 @@
+const knex = require('knex');
+const knexConfig = require('../knexfile.js');
+
+db = knex(knexConfig.development);
+
+module.exports = {
+    getDishes,
+    addDish,
+    getDish,
+    getRecipes,
+    addRecipe
+}
+
+function getDishes() {
+    return db('dishes');
+  }
+  
+  function getDish(id) { // include list of related recipes
+    return db('dishes').where("dishes.id",id)
```

Comment:
I believe this is where your issue is when trying to access the dishes by id. Check out this syntax and compare it to what Luis did in lecture.

Reply:
Thanks for feedback! I forgot to provide `.length`  property to my get by id dishes endpoint here: `server.get('/api/dishes/:id', (req, res) => {
    const id = req.params.id;
    db.getDish(id)
    .then(response => {
        if(response.length > 0){
            res.status(200).json(response)
        } else {
            res.status(404).json({ error: "There is no Dish with the provided id" });
        }
    })
    .catch(err => {
        console.log(err)
        res.status(500).json(err)
    })
})`
Now, I think the issue is fixed.